components/ai_chat/core/browser/engine/conversation_api_client.cc

```diff
@@ -217,14 +218,15 @@ void ConversationAPIClient::ClearAllQueries() {
 
 void ConversationAPIClient::PerformRequest(
     std::vector<ConversationEvent> conversation,
+    const EngineConsumer::Tools tools,
     const std::string& selected_language,
     GenerationDataCallback data_received_callback,
     GenerationCompletedCallback completed_callback,
     const std::optional<std::string>& model_name) {
   // Get credentials and then perform request
   auto callback = base::BindOnce(
       &ConversationAPIClient::PerformRequestWithCredentials,
-      weak_ptr_factory_.GetWeakPtr(), std::move(conversation),
+      weak_ptr_factory_.GetWeakPtr(), std::move(conversation), tools,
```

Comment:
As mentioned in the DM, it isn't safe to bind the vector of pointers here.

Reply:
removed from this PR, we can discuss in a subsequent one